pandas-stubs/io/excel/_base.pyi

```diff
@@ -31,6 +31,86 @@ from pandas._typing import (
+# 1 and 3 overlap with incompatible return types
+def read_excel(  # type:ignore[misc]
+    io: FilePath
+    | ReadBuffer[bytes]
+    | bytes
+    | ExcelFile
+    | Workbook
+    | Book
+    | OpenDocument
+    | pyxlsb.workbook.Workbook,
+    sheet_name: list[int],
+    *,
+    header: int | Sequence[int] | None = ...,
+    names: ListLikeHashable | None = ...,
+    index_col: int | Sequence[int] | None = ...,
+    usecols: str | UsecolsArgType = ...,
+    dtype: str | Dtype | Mapping[str, str | Dtype] | None = ...,
+    engine: Literal["xlrd", "openpyxl", "odf", "pyxlsb"] | None = ...,
+    converters: Mapping[int | str, Callable[[object], object]] | None = ...,
+    true_values: Iterable[Hashable] | None = ...,
+    false_values: Iterable[Hashable] | None = ...,
+    skiprows: int | Sequence[int] | Callable[[object], bool] | None = ...,
+    nrows: int | None = ...,
+    na_values: Sequence[str] | dict[str | int, Sequence[str]] = ...,
+    keep_default_na: bool = ...,
+    na_filter: bool = ...,
+    verbose: bool = ...,
+    parse_dates: bool
+    | Sequence[int]
+    | Sequence[Sequence[str] | Sequence[int]]
+    | dict[str, Sequence[int] | list[str]] = ...,
+    date_format: dict[Hashable, str] | str | None = ...,
+    thousands: str | None = ...,
+    decimal: str = ...,
+    comment: str | None = ...,
+    skipfooter: int = ...,
+    storage_options: StorageOptions = ...,
+    dtype_backend: DtypeBackend | NoDefault = ...,
+) -> dict[int, DataFrame]: ...
+@overload
+# 2 and 3 overlap with incompatible return types
+def read_excel(  # type:ignore[misc]
+    io: FilePath
+    | ReadBuffer[bytes]
+    | bytes
+    | ExcelFile
+    | Workbook
+    | Book
+    | OpenDocument
+    | pyxlsb.workbook.Workbook,
+    sheet_name: list[str] | None,
+    *,
+    header: int | Sequence[int] | None = ...,
+    names: ListLikeHashable | None = ...,
+    index_col: int | Sequence[int] | None = ...,
+    usecols: str | UsecolsArgType = ...,
+    dtype: str | Dtype | Mapping[str, str | Dtype] | None = ...,
+    engine: Literal["xlrd", "openpyxl", "odf", "pyxlsb"] | None = ...,
+    converters: Mapping[int | str, Callable[[object], object]] | None = ...,
+    true_values: Iterable[Hashable] | None = ...,
+    false_values: Iterable[Hashable] | None = ...,
+    skiprows: int | Sequence[int] | Callable[[object], bool] | None = ...,
+    nrows: int | None = ...,
+    na_values: Sequence[str] | dict[str | int, Sequence[str]] = ...,
+    keep_default_na: bool = ...,
+    na_filter: bool = ...,
+    verbose: bool = ...,
+    parse_dates: bool
+    | Sequence[int]
+    | Sequence[Sequence[str] | Sequence[int]]
+    | dict[str, Sequence[int] | list[str]] = ...,
+    date_format: dict[Hashable, str] | str | None = ...,
+    thousands: str | None = ...,
+    decimal: str = ...,
+    comment: str | None = ...,
+    skipfooter: int = ...,
+    storage_options: StorageOptions = ...,
+    dtype_backend: DtypeBackend | NoDefault = ...,
+) -> dict[str, DataFrame]: ...
```

Comment:
Can the first two overloads be combined using a _StrIntT TypeVar? (I think we might even have such a typevar in _typing)

Reply:
pandas does that: https://github.com/pandas-dev/pandas/blob/9e1096e8373bc99675fd1b3490cfb7cf26041395/pandas/io/excel/_base.py#L426

Reply:
I've put that in.  Still need 4 overloads because `sheetname = None` gives a `dict[str, pd.DataFrame]` result.